Docs update for includeFiles and excludeFiles options on netlify adapter

dfdez opened the pull request:
**Description:**
Docs update for https://github.com/withastro/astro/pull/13194.

> [!NOTE]  
> This PR remains in draft status pending the merge of these options into the adapter. I will mark it as ready for review once the changes are deployed to production.

astrobot-houston commented:
<!-- lunaria-action-comment -->

## Lunaria Status Overview


🌕 **This pull request will trigger status changes.**

<details>
<summary>Learn more</summary>

By default, every PR changing files present in the [Lunaria configuration's `files` property](https://lunaria.dev/reference/configuration/#files-required) will be considered and trigger status changes accordingly.

You can change this by adding one of the keywords present in the [`ignoreKeywords` property in your Lunaria configuration file](https://lunaria.dev/reference/configuration/#ignorekeywords) in the PR's title (ignoring all files) or by [including a tracker directive](https://lunaria.dev/guides/tracking/#tracker-directives) in the merged commit's description. 
</details>

### Tracked Files




| File                                                                                                                                                                                              | Note                                                       |
| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------- |
| [en/guides/integrations-guide/netlify.mdx](https://github.com/withastro/docs/blob/08b551936f5144b9b907c61f3ac2e08bad9bccd4/src%2Fcontent%2Fdocs%2Fen%2Fguides%2Fintegrations-guide%2Fnetlify.mdx) | Source changed, localizations will be marked as outdated.  |
| [es/guides/integrations-guide/netlify.mdx](https://github.com/withastro/docs/blob/08b551936f5144b9b907c61f3ac2e08bad9bccd4/src%2Fcontent%2Fdocs%2Fes%2Fguides%2Fintegrations-guide%2Fnetlify.mdx) | Localization changed, will be marked as complete.          |

<details>
<summary>Warnings reference</summary>

| Icon | Description                                                                                                                                       |
| ---- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
| 🔄️  | The source for this localization has been updated since the creation of this pull request, make sure all changes in the source have been applied. |

</details>

astrobot-houston commented:
Hello! Thank you for opening your **first PR** to Astro’s Docs! 🎉

Here’s what will happen next:

1. Our GitHub bots will run to check your changes.
   If they spot any broken links you will see some error messages on this PR.
   Don’t hesitate to ask any questions if you’re not sure what these mean!

2. In a few minutes, you’ll be able to see a preview of your changes on Netlify 🥳.

3. One or more of our maintainers will take a look and may ask you to make changes.
   We try to be responsive, but don’t worry if this takes a few days.

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
```diff
@@ -244,6 +244,39 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
 standard caching headers like `CDN-Cache-Control` or `Vary`.
 Refer to the docs to learn about implementing e.g. time to live (TTL) or stale while revalidate (SWR) caching: https://docs.netlify.com/platform/caching
 
+### Including files on SSR
```
```suggestion
### Including or excluding files
```
The section is about both!

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
```diff
@@ -244,6 +244,39 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
 
+### Including files on SSR
+
+When using Server-Side Rendering (SSR), you may need to include specific files in your Netlify Functions that aren't automatically bundled during deployment.
```
```suggestion
You may need to include specific files in your Netlify Functions for on-demand rendering that aren't automatically bundled during deployment.
```

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
```diff
@@ -244,6 +244,39 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
+### Including files on SSR
+
+When using Server-Side Rendering (SSR), you may need to include specific files in your Netlify Functions that aren't automatically bundled during deployment.
+The `includeFiles` property allows you to explicitly specify which files should be bundled with your function.
```
```suggestion
The `includeFiles` property allows you to explicitly specify additional files that should be bundled with your function.
```
You don't specify *all the* files, just *additional* ones, right? I think this is more accurate!

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
````diff
@@ -244,6 +244,39 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
+The `includeFiles` property allows you to explicitly specify which files should be bundled with your function.
+
+```js title="astro.config.mjs" ins={8}
+import { defineConfig } from 'astro/config';
+import vercel from '@astrojs/vercel';
+
+export default defineConfig({
+  // ...
+  output: 'server',
+  adapter: vercel({
+    includeFiles: ['./my-data.json'],
+  }),
+});
````
```suggestion
import { defineConfig } from 'astro/config';
import netlify from '@astrojs/netlify';

export default defineConfig({
  // ...
  output: 'server',
  adapter: netlify({
    includeFiles: ['./my-data.json'],
  }),
});
```
Should this be `netlify`?

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
````diff
@@ -244,6 +244,39 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
+You can also add the `excludeFiles` property to exclude any files from the bundling process that would otherwise be included.
+
+```js title="astro.config.mjs" ins={8}
+import { defineConfig } from 'astro/config';
+import vercel from '@astrojs/vercel';
+
+export default defineConfig({
+  // ...
+  output: 'server',
+  adapter: vercel({
+    excludeFiles: ['./src/some_big_file.jpg'],
+  }),
+});
````
```suggestion
import { defineConfig } from 'astro/config';
import netlify from '@astrojs/netlify';

export default defineConfig({
  // ...
  output: 'server',
  adapter: netlify({
    excludeFiles: ['./src/some_big_file.jpg'],
  }),
});
```
`netlify`?

sarah11918 reviewed (commented):
Thank you for submitting these docs for your feature! I've made a few suggestions below to conform to our style and language use. And, I think the examples should all refer to `netlify` here and not `vercel`, right? 😄 

Also, please remove the translation file. We have a process for translation, and our PR template reminds you that you should only make a PR in one language. Thank you for following our system!

I'm looking forward to having this feature in docs! 

For Team Docs looking at this PR: this PR highlights that the Netlify and Vercel docs are structured a little differently. (The Vercel documentation lists these as configuration options, whereas the Netlify docs are more "topic based." We can't fix that in *this* PR, but we may want to eventually move to consistency between the two! 😅

dfdez commented:
Hey @sarah11918 thanks for the feedback!

I've updated the examples to reference Netlify instead of Vercel and removed the translation file as requested. 

I've also added more information to the docs as requested. Let me know if there's anything else needed 👍🏽

sarah11918 reviewed (commented):
Just a quick note these are in really good shape, and I'm waiting on the implementation PR to be reviewed/approved (just to make sure there's no new info/nothing changes) before doing a final polish here! 🫡

sarah11918 commented:
Also noting that now that we've moved the adapters, this *shouldn't* affect the docs we've got going here, but we *will* need to update the PR description so we can link to the new PR to the main Astro repo instead of the closed adapters repo PR!  🙌

sarah11918 commented:
Thanks for updating the description to link to the new `withastro/astro` PR!

Reminder (for all of us) that these docs will then still need final polishing once they are out of draft, when I'll compare to the implementation PR itself, the changeset there etc.!  🙌

dfdez commented:
Important [comment](https://github.com/withastro/astro/pull/13194#discussion_r1949124541) made on the implementation PR that we should consider for the docs to 👍

ematipico commented:
@sarah11918 I added the relative code PR to the next release 5.3.0

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
```diff
@@ -244,6 +244,75 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
+
+When deploying an Astro site with on-demand rendering to Netlify, the generated functions automatically trace and include server dependencies. However, you may need to customize which files are included in your Netlify Functions.
+
+#### Including additional files
```
```suggestion
#### `includeFiles`

<p>

**Type:**  `string[]`<br />
**Default:** `[]`<br />
<Since v="5.3.0" />
</p>
```
Let's at least start updating this page to our "new" normal API standards by including this API reference syntax, which also allows us to include an "Added in". (Make this accurate!)

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
````diff
@@ -244,6 +244,75 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
+});
+```
+
+#### Excluding files
````
```suggestion
#### `excludeFiles`

<p>

**Type:**  `string[]`<br />
**Default:** `[]`<br />
<Since v="5.3.0" />
</p>
```

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
```diff
@@ -244,6 +244,75 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
 standard caching headers like `CDN-Cache-Control` or `Vary`.
 Refer to the docs to learn about implementing e.g. time to live (TTL) or stale while revalidate (SWR) caching: https://docs.netlify.com/platform/caching
 
+### Including or excluding files
```
```suggestion
### Including or excluding files from Netlify Functions
```

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
```diff
@@ -244,6 +244,75 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
+- Data files loaded using `fs` operations
+- Configuration files
+- Template files
+
```
```suggestion

Provide an array of additional files to include with file paths relative to your project's [`root`](/en/reference/configuration-reference/#root). Absolute paths may not work as expected.

```

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
```diff
@@ -244,6 +244,75 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
+  // ...
+  output: 'server',
+  adapter: netlify({
+    includeFiles: ['./my-data.json'],
```
```suggestion
    includeFiles: ['./my-data.json'], // relative to `root`
```

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
```diff
@@ -244,6 +244,75 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
+- Reducing bundle size
+- Excluding large binaries
+- Preventing unwanted files from being deployed
+
```
```suggestion

Provide an array of specific files to exclude with file paths relative to your project's [`root`](/en/reference/configuration-reference/#root). Absolute paths may not work as expected.

```

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
```diff
@@ -244,6 +244,75 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
+  // ...
+  output: 'server',
+  adapter: netlify({
+    excludeFiles: ['./src/some_big_file.jpg'],
```
```suggestion
    excludeFiles: ['./src/some_big_file.jpg'], // relative to `root`
```

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
````diff
@@ -244,6 +244,75 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
+  }),
+});
+```
+
+Remember that paths should be relative to your project root (start with `./`).
````
```suggestion
```

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
```diff
@@ -244,6 +244,75 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
+
+#### Using glob patterns
+
+Both `includeFiles` and `excludeFiles` support glob patterns for matching multiple files:
```
```suggestion
Both `includeFiles` and `excludeFiles` support [glob patterns](/en/guides/imports/#glob-patterns) for matching multiple files:
```

sarah11918 reviewed (commented):
Looks great! Just some polishing changes, including making the additions more like our standard API reference. (I know, this is an older page that hasn't yet been updated in other entries, but let's at least get this one in shape!)

Please do check for accuracy!

dfdez commented:
Hey @sarah11918 thanks for the feedback again!

I have made a [commit](https://github.com/withastro/docs/pull/10760/commits/8bae00cdbb2bb32f3683c12fe900e2cafa7a234c) including all the suggested changes 👌

sarah11918 commented on src/content/docs/en/guides/integrations-guide/netlify.mdx:
````diff
@@ -244,6 +245,89 @@ With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grai
+});
+```
+
+Provide an array of specific files to exclude with file paths relative to your project's [`root`](/en/reference/configuration-reference/#root). Absolute paths may not work as expected.
+
+#### `excludeFiles`
+
+<p>
+**Type:**  `string[]`<br />
+**Default:** `[]`<br />
+<Since v="5.3.0" />
+</p>
+
+You can use the `excludeFiles` property to prevent specific files from being bundled that would otherwise be included. This is helpful for:
+- Reducing bundle size
+- Excluding large binaries
+- Preventing unwanted files from being deployed
+
````
```suggestion
#### `excludeFiles`

<p>
**Type:**  `string[]`<br />
**Default:** `[]`<br />
<Since v="5.3.0" />
</p>

You can use the `excludeFiles` property to prevent specific files from being bundled that would otherwise be included. This is helpful for:
- Reducing bundle size
- Excluding large binaries
- Preventing unwanted files from being deployed

Provide an array of specific files to exclude with file paths relative to your project's [`root`](/en/reference/configuration-reference/#root). Absolute paths may not work as expected.

```
This just ended up in the wrong place!

sarah11918 reviewed (approved):
Thanks so much for the updates, and thanks for also importing that `<Since />` component! Welcome to Team Docs! 🥳
